### ISSUE TITLE
Add example to test_ui using a game_t struct with initialized rooms and fix start_ui() seg faults

### ISSUE THREAD

**Issue by omorkved** [uchicago-cs/chiventure#264]
Right now, examples/test_ui.c creates a context from a game struct that has no initialized rooms, which causes start_ui.c to have a lot of errors. Thus, to avoid errors we have currently set  ui_ctx->coord_hash = get_test_coord_hash(), a mock coordinate hash, inside of the initialization function in ui_ctx.c to produce a pre-set hash. This task is to create some rooms using game state functions, get test_ui to run, and fix the seg faults that will arise in re-setting the coord_hash to: ui_ctx->coord_hash= create_valid_map(game)

**Comment by omorkved** [comment]
This task has been on hold as we wait for #263 to go through. In the meantime, I've created the test game:

`// Sample Rooms                                                                                                                     
    room_t *room1 = room_new("One", "", "");
    room_t *room2 = room_new("Two", "", "");
    room_t *room3 = room_new("Three", "", "");
    room_t *room4 = room_new("Four", "", "");
    room_t *room5 = room_new("Five", "", "");

    chiventure_ctx_t *ctx = chiventure_ctx_new();

    add_room_to_game(ctx->game, room1);
    add_room_to_game(ctx->game, room2);
    add_room_to_game(ctx->game, room3);
    add_room_to_game(ctx->game, room4);
    add_room_to_game(ctx->game, room5);
    create_connection(ctx->game, "One", "Two", "east");
    create_connection(ctx->game, "Two", "One", "west");
    create_connection(ctx->game, "Two", "Three", "south");
    create_connection(ctx->game, "Three", "Two", "north");
    create_connection(ctx->game, "Five", "Three", "north");
    create_connection(ctx->game, "Three", "Five", "south");`

which I've spliced into a second map example that reads much like the map_example.c file. So for, so good, but waiting on changes in that PR. This file will be committed in a new branch spun off master the second the PR goes through.

**Comment by omorkved** [comment]
I'm thinking this example above might not get into master, but I'd like to at least push a hotfix soon (i.e. today) to set ui_ctx->coord_hash= create_valid_map(game) so that we can _actually_ generate a map from Game State data. (delayed due to PR above)

**Comment by omorkved** [comment]
#348 is the implementation of ui_ctx->coord_hash= create_valid_map(game) !!

**Comment by omorkved** [comment]
Summary: our DFS algorithm is now called in create_valid_map()! However, it has never been called extensively on larger files (say, 20+ rooms) so we don't know if it is fully working (future easter egg for next year's team??). That being said, even when create_valid_map() fails, I added in a few extra things explained in #348 guaranteeing that we at least default to printing a test map whenever DFS fails (as opposed to throwing segfaults for files with invalid maps). (Check #348 for more written info on this)

The second map example was never quite finished because pull requests got crazy in the past 72 hours including some 2am last-minute hotfixes, and creating an example was lower priority than making sure all updates with game-state, cli, etc. continued to work and that we got the 3-4 UI PR's created in the last 72 hours into master. So that remains unfinished, but extensive testing will have to be left to a future project development team !! :)
